stdlib/asm/crypto/stark/public_inputs.masm

```diff
@@ -81,13 +81,153 @@ export.load
         end
 
         # Absorb the last digest and pad with zeros
-        adv_loadw
-        swapw
-        exec.constants::zeroize_stack_word
+        exec.load_base_store_extension_word
         hperm
     end
 
     exec.rpo::squeeze_digest
     movup.4 drop
     movup.4 drop
 end
+
+#! Computes the address where the public inputs are to be stored.
+#!
+#! In order to be able to call `arithmetic_circuit_eval`, we need to layout the inputs to
+#! the constraint evaluation circuit in a contiguous region of memory (called `READ` section
+#! in the ACE chiplet documentation) right before the region of memory storing the circuit
+#! description (called `EVAL` section in the ACE chiplet documentation).
+#! As the number of public inputs is a per-instance parameter, while the sizes of the OOD
+#! evaluation frames and the number of auxiliary random values are fixed, we can lay out
+#! the public inputs right before the auxiliary random values and OOD evaluations.
+#! Hence the address where public inputs are stored is computed using a negative offset
+#! from the address where the OOD are stored.
+#!
+#! Input: [...]
+#! Output: [...]
```

Comment:
This also returns the public inputs address, right? Then we should update the `Output` section

Reply:
The `mem_store` pops the value from the stack and hence the return stack is empty, or was it something else?

Reply:
Does [this](https://github.com/0xMiden/miden-vm/blob/179edd11ae63d13f7174cbd597e2cf151eb107ce/stdlib/asm/crypto/stark/public_inputs.masm#L128) `dup` make the stack before `mem_store` look like `[public_inputs_address_ptr, offset, offset]`? Which after `mem_store` is `[offset]`, and then returned?

If not, can you add comments in the code to track the stack? Clearly I'm missing something in that case.

Reply:
Sorry about that, I totally missed that dup. Fixed the docs and the output stack description. Thanks for catching it